transformations/transformations/pool_allocator.py

```diff
@@ -477,10 +600,10 @@ def create_pool_allocator(self, routine, stack_size):
             # Find OpenMP parallel statements
             omp_pragmas = [p for p in FindNodes(Pragma).visit(routine.body) if p.keyword.lower() == 'omp']
             for pragma in omp_pragmas:
-                if pragma.content.startswith('parallel'):
+                if pragma.content.lower().startswith('parallel'):
```

Comment:
It feels like working with Pragmas is unnecessarily clumsy right now. If you have any suggestions for better utilities/additions to make this a little smoother? Doesn't have to happen in this PR but maybe worth opening an issue about it

Reply:
Yes I agree, with the nested ifs and `lower()` all over the place its not particularly readable. I'll open an issue.